common/log/class.SingleFileAppender.php

```diff
@@ -136,11 +136,17 @@ protected function initFile()
     {
         if ($this->maxFileSize > 0 && file_exists($this->filename) && filesize($this->filename) >= $this->maxFileSize) {
             // need to reduce the file size
-            $file = file($this->filename);
-            $file = array_splice($file, ceil(count($file) * $this->reduceRatio));
+            $reduceLen= ($this->maxFileSize< (int)ini_get('memory_limit')*1024?$this->maxFileSize:(int)ini_get('memory_limit')*1024)*$this->reduceRatio;
```

Comment:
For readability it might be better to change this from the shorthand if else to the full controll structure:

```
if ($this->maxFileSize < (int)ini_get('memory_limit') * 1024) {
    $reduceLength = ($this->maxFileSize) * $this->reduceRatio;
} else {
    $reduceLength = ((int)ini_get('memory_limit') * 1024) * $this->reduceRatio;
}
```

Reply:
A variable for `(int)ini_get('memory_limit') * 1024` wouldn't also be out of place here